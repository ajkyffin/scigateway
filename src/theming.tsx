import { createMuiTheme } from '@material-ui/core/styles';
import { ThemeOptions, Theme } from '@material-ui/core/styles/createMuiTheme';

export interface UKRIThemeOptions extends ThemeOptions {
  ukri: {
    purple: string;
    blue: string;
    orange: string;
    green: string;
    grey: string;
  };
}

export interface UKRITheme extends Theme {
  ukri: {
    purple: string;
    blue: string;
    orange: string;
    green: string;
    grey: string;
  };
}

export const buildTheme = (): Theme => {
  const options: UKRIThemeOptions = {
    typography: {
      useNextVariants: true,
    },
    palette: {
      primary: {
        main: '#1D4F91', // dark blue
      },
      secondary: {
        main: '#63666A', //grey
      },
    },
    ukri: {
      purple: '#8C4799',
      blue: '#1D4F91',
      orange: '#C34613',
      green: '#008275',
      grey: '#63666A',
    },
  };

  return createMuiTheme(options);
};
