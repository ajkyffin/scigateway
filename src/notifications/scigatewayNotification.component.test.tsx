import React from 'react';
import { createShallow, createMount } from '@material-ui/core/test-utils';
import { NotificationWithStyles } from './scigatewayNotification.component';
import { Action } from 'redux';
import { buildTheme } from '../theming';
import { MuiThemeProvider } from '@material-ui/core';

const theme = buildTheme();

function createScigatewayNotification(
  severity: string,
  message: string
): React.ReactElement {
  return (
    <MuiThemeProvider theme={theme}>
      <NotificationWithStyles
        message={message}
        severity={severity}
        index={0}
        dismissNotification={(): Action => ({ type: 'test' })}
      />
    </MuiThemeProvider>
  );
}

describe('Scigateway Notification component', () => {
  let shallow;
  let mount;

  beforeEach(() => {
    shallow = createShallow({ untilSelector: 'div' });
    mount = createMount();
  });

  afterEach(() => {
    mount.cleanUp();
  });

  it('Scigateway Notification success message renders correctly', () => {
    const wrapper = shallow(
      createScigatewayNotification('success', 'success message')
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('Scigateway Notification warning message renders correctly', () => {
    const wrapper = shallow(
      createScigatewayNotification('warning', 'warning message')
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('Scigateway Notification error message renders correctly', () => {
    const wrapper = shallow(
      createScigatewayNotification('error', 'error message')
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('an action is fired when Scigateway Notification button is clicked', () => {
    const mockDismissFn = jest.fn();

    const wrapper = mount(
      <MuiThemeProvider theme={theme}>
        <NotificationWithStyles
          message={'warning message'}
          severity={'warning'}
          index={0}
          dismissNotification={mockDismissFn}
        />
      </MuiThemeProvider>
    );

    wrapper.find('button').simulate('click');

    expect(mockDismissFn.mock.calls.length).toEqual(1);
  });
});
