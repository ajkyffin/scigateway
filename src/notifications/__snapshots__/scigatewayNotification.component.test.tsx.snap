// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scigateway Notification component Scigateway Notification error message renders correctly 1`] = `
<div
  className="root-class"
>
  <HighlightOffIcon
    className="errorIcon-class"
  />
  <WithStyles(ForwardRef(Typography))
    variant="body2"
  >
    error message
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(IconButton))
    aria-label="Dismiss notification"
    className="button-class"
    onClick={[Function]}
  >
    <ClearIcon
      className="deleteIcon-class"
    />
  </WithStyles(ForwardRef(IconButton))>
</div>
`;

exports[`Scigateway Notification component Scigateway Notification success message renders correctly 1`] = `
<div
  className="root-class"
>
  <CheckCircleIcon
    className="successIcon-class"
  />
  <WithStyles(ForwardRef(Typography))
    variant="body2"
  >
    success message
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(IconButton))
    aria-label="Dismiss notification"
    className="button-class"
    onClick={[Function]}
  >
    <ClearIcon
      className="deleteIcon-class"
    />
  </WithStyles(ForwardRef(IconButton))>
</div>
`;

exports[`Scigateway Notification component Scigateway Notification warning message renders correctly 1`] = `
<div
  className="root-class"
>
  <ErrorOutlineIcon
    className="warningIcon-class"
  />
  <WithStyles(ForwardRef(Typography))
    variant="body2"
  >
    warning message
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(IconButton))
    aria-label="Dismiss notification"
    className="button-class"
    onClick={[Function]}
  >
    <ClearIcon
      className="deleteIcon-class"
    />
  </WithStyles(ForwardRef(IconButton))>
</div>
`;
